services/proxy/pkg/staticroutes/backchannellogout.go

```diff
@@ -62,3 +71,42 @@ func (s *StaticRouteHandler) backchannelLogout(w http.ResponseWriter, r *http.Re
 	render.Status(r, http.StatusOK)
 	render.JSON(w, r, nil)
 }
+
+// publishBackchannalLogoutEvent publishes a backchannel logout event when a user logs out
+func (s StaticRouteHandler) publishBackchannalLogoutEvent(ctx context.Context, record *microstore.Record, logoutToken *oidc.LogoutToken) error {
+	urecords, err := s.UserInfoCache.Read(string(record.Value))
+	if err != nil {
+		return fmt.Errorf("reading userinfo cache: %w", err)
+	}
+	if len(urecords) == 0 {
+		return fmt.Errorf("userinfo not found")
+	}
+
+	var claims map[string]interface{}
+	if err = msgpack.UnmarshalAsMap(urecords[0].Value, &claims); err != nil {
+		return fmt.Errorf("could not unmarshal userinfo: %w", err)
+	}
+
+	oidcClaim, ok := claims[s.Config.UserOIDCClaim].(string)
+	if !ok {
+		return fmt.Errorf("could not get claim %w", err)
+	}
+
+	user, _, err := s.UserProvider.GetUserByClaims(ctx, s.Config.UserCS3Claim, oidcClaim)
+	if err != nil || user.GetId() == nil {
+		return fmt.Errorf("could not get user by claims: %w", err)
+	}
+
+	e := events.BackchannelLogout{
+		Executant: user.GetId(),
+		SessionId: logoutToken.SessionId,
+		Timestamp: utils.TSNow(),
+	}
+
+	if s.EventsPublisher != nil {
```

Comment:
This is no good check. We don't need to do anything in this function if we don't have an `EventsPublisher`. Either check it at the beginning of the function and return directly OR check it during initialization and assume it is `!= nil` here. I'd go with second solution, but first one also fine if the publisher is optional.

Reply:
Do you mean during the proxy service initialization?
If yes I don't think this is a good idea. Sounds to me like we should not interrupt the proxy launching